packages/kbn-es-query/src/es_query/types.ts

```diff
@@ -34,7 +36,8 @@ export interface IFieldSubTypeNested {
  * A base interface for an index pattern field
  * @public
  */
-export interface DataViewFieldBase {
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type DataViewFieldBase = {
```

Comment:
These changes are necessary for `DataViewFieldBase` to meet the `SerializableRecord` definition, which is necessary for implementing persistable state. I could also do `export interface DataViewFieldBase extends SerializableRecord` but that add unnamed properties to the `DataViewFieldBase` spec.